csrc/options.h

```diff
@@ -86,6 +86,7 @@ enum class EnableOption {
   MemoryPromotion, //! Enable promotion of memory types for non-pointwise ops
   StaticFusionCount, //! Enable using single static count in kernel name
   WarnRegisterSpill, //! Enable warnings of register spill
+  MatmulExprEval, //! Enable ATen evaluation for Matmul
```

Comment:
I think we need to make a louder statement here regarding the impact of this flag.

Enabling `MatmulExprEval` means that we are running the entire fusion containing a matmul with expression evaluation, not just the matmul portion.